bigquery/schema_test.go

```diff
@@ -1204,3 +1233,66 @@ func TestSchemaFromJSON(t *testing.T) {
 		}
 	}
 }
+
+func TestFormatJSONFields(t *testing.T) {
+
+	// cmp option for comparing byte arrays without caring about whitespace.
+	// courtesy of https://github.com/google/go-cmp/issues/224
+	normalizeJSON := cmp.FilterValues(func(x, y []byte) bool {
+		return json.Valid(x) && json.Valid(y)
+	}, cmp.Transformer("ParseJSON", func(in []byte) (out interface{}) {
+		if err := json.Unmarshal(in, &out); err != nil {
+			panic(err)
+		}
+		return out
+	}))
+
+	testCases := []struct {
+		description  string
+		inSchema     Schema
+		expectedJSON []byte
+	}{
+		{
+			description: "basic schema",
+			inSchema: Schema{
+				fieldSchema("foo", "strfield", "STRING", false, false, nil),
+				fieldSchema("bar", "intfield", "INTEGER", false, true, nil),
+				fieldSchema("baz", "bool_arr", "INTEGER", true, false, []string{"tag1"}),
```

Comment:
What happens if both required and repeated are true?

Reply:
Disallowed.  `Required` is ignored if `Repeated` is true.  See FieldSchema comments: https://pkg.go.dev/cloud.google.com/go/bigquery#FieldSchema

Under the hood this is controlled by a single "mode" value: required, optional, repeated.